test/pjrt/test_runtime.py

```diff
@@ -55,24 +52,21 @@ def test_xla_device_error(self):
   }, False), ('pjrt_cpu', {
       'PJRT_DEVICE': 'CPU',
       'PJRT_SELECT_DEFAULT_DEVICE': '0'
-  }, True), ('xrt_tpu', {
-      'XRT_TPU_CONFIG': 'localservice;0;localhost:51011'
-  }, False), ('pjrt_tpu_precedence', {
+  }, True), ('pjrt_tpu_precedence', {
       'PJRT_DEVICE': 'TPU',
       'XRT_TPU_CONFIG': 'localservice;0;localhost:51011',
-  }, True), ('xrt_gpu', {
-      'GPU_NUM_DEVICES': '4'
-  }, False), ('pjrt_gpu', {
+  }, True), ('pjrt_gpu', {
       'PJRT_DEVICE': 'GPU',
       'GPU_NUM_DEVICES': '4'
   }, True), ('xla_dist_worker', {
-      'XRT_LOCAL_WORKER': 'c_localservice:2'
-  }, False))
+      'PJRT_DEVICE': 'TPU',
+  }, True))
   def test_pjrt_default_device(self, env_vars, expect_using_pjrt):
     with mock.patch.dict(os.environ, env_vars, clear=True):
       # Print a warningif we had to select a default runtime
       if 'PJRT_DEVICE' not in os.environ and expect_using_pjrt:
         logs_context = self.assertLogs(level=logging.WARNING)
+        print("XRT has been deprecated, please set PJRT_DEVICE")
```

Comment:
you should just error out here, PJRT will become the only runtime, not setting it is an error

Reply:
thanks for pointing, test in https://github.com/pytorch/xla/pull/5223

Reply:
try to raise an error here, but failed in https://github.com/pytorch/xla/pull/5223/commits/cd0f293c586e55e75cd3b573241d3ca9533cd2b0

since this is a test to test pjrt,  we have several situation here for PJRT test:
-   ('default', /*env_vars=*/{}, /*expect_using_pjrt=*/True), 
  - this set is testing whether we have PJRT as default
-   ('no_default', /*env_vars=*/{'PJRT_SELECT_DEFAULT_DEVICE': '0'}, /*expect_using_pjrt=*/False), 
  - this set is no PJRT_DEVICE set, and no device get
-  ('pjrt_cpu', /*env_vars=*/{'PJRT_DEVICE': 'CPU', 'PJRT_SELECT_DEFAULT_DEVICE': '0'}, /*expect_using_pjrt=*/True), 
  - PJRT_DEVICE=CPU test, will keep
-  ('xrt_tpu', /*env_vars=*/{'XRT_TPU_CONFIG': 'localservice;0;localhost:51011'}, /*expect_using_pjrt=*/False), 
  - this set has been deleted, no XRT code and no XRT TPU test
-  ('pjrt_tpu_precedence', /*env_vars=*/{'PJRT_DEVICE': 'TPU', 'XRT_TPU_CONFIG': 'localservice;0;localhost:51011',}, /*expect_using_pjrt=*/True), 
  - PJRT_DEVICE=TPU test, will keep
-  ('xrt_gpu', /*env_vars=*/{'GPU_NUM_DEVICES': '4'}, /*expect_using_pjrt=*/False), 
  - this set has been delelted, no XRT code and no XRT GPU test
-  ('pjrt_gpu', /*env_vars=*/{'PJRT_DEVICE': 'GPU','GPU_NUM_DEVICES': '4'}, /*expect_using_pjrt=*/True), 
  - PJRT_DEVICE=GPU test, will keep
-  ('xla_dist_worker', /*env_vars=*/{'XRT_LOCAL_WORKER': 'c_localservice:2'}, /*expect_using_pjrt=*/False))
  - this set will be delete, `xla_dist` is for XRT, and will delete XRT's xla_dist test


for situation which PJRT is not set and still want PJRT, we will set default PJRT_DEVICE to `CPU` or `TPU` based on whether or not found libtpu.so and TPU device: https://github.com/pytorch/xla/blob/b3f4c7bc1e9f812069770c83e38e5bd2dc20aef6/torch_xla/runtime.py#L36C1-L51C41

so we should not raise ERROR here for each situation

Reply:
Thanks, updated by deleting this print comments